flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistryImpl.java

```diff
@@ -147,14 +145,18 @@ public MetricRegistryImpl(MetricRegistryConfiguration config, Collection<Reporte
 					} else {
 						LOG.info("Reporting metrics for reporter {} of type {}.", namedReporter, className);
 					}
-					reporters.add(new ReporterAndSettings(reporterInstance, new ReporterScopedSettings(reporters.size() - 1)));
 
 					String delimiterForReporter = reporterSetup.getDelimiter().orElse(String.valueOf(globalDelimiter));
 					if (delimiterForReporter.length() != 1) {
 						LOG.warn("Failed to parse delimiter '{}' for reporter '{}', using global delimiter '{}'.", delimiterForReporter, namedReporter, globalDelimiter);
 						delimiterForReporter = String.valueOf(globalDelimiter);
 					}
-					this.delimiters.add(delimiterForReporter.charAt(0));
+
+					reporters.add(new ReporterAndSettings(
+						reporterInstance,
+						new ReporterScopedSettings(
+							reporters.size() - 1,
```

Comment:
Is this index calculation correct? What if we are adding the first reporter. Wouldn't the index then be `-1`?

Reply:
If this is incorrect, then I would suggest to add a test to guard against future regressions.